### ISSUE TITLE
[java] UselessQualifiedThis - false positive with 'this' in a lambda pointing to the instance outside the lambda

### ISSUE THREAD

**Issue by garydgregory** [pmd/pmd#4030]
**Affects PMD Version:** 6.47.0

**Rule:** UselessQualifiedThis

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_codestyle.html#uselessqualifiedthis>

**Description:**

**Code Sample demonstrating the issue:**

```
    private void addDefinitionBuilderChangeListener(final ConfigurationBuilder<? extends HierarchicalConfiguration<?>> defBuilder) {
        defBuilder.addEventListener(ConfigurationBuilderEvent.RESET, event -> {
            synchronized (CombinedConfigurationBuilder.this) {
                reset();
                definitionBuilder = defBuilder;
            }
        });
    }
```

See https://github.com/apache/commons-configuration/blob/f87adc117fe00afacfdad16d825257abc9ed1571/src/main/java/org/apache/commons/configuration2/builder/combined/CombinedConfigurationBuilder.java#L1023

**Expected outcome:**

PMD reports a violation at line 1023, but that's wrong. That's a false positive.
```
<file name="/Users/garydgregory/git/commons-configuration/src/main/java/org/apache/commons/configuration2/builder/combined/CombinedConfigurationBuilder.java">
<violation beginline="1023" endline="1023" begincolumn="27" endcolumn="54" rule="UselessQualifiedThis" ruleset="Code Style" package="org.apache.commons.configuration2.builder.combined" class="CombinedConfigurationBuilder" method="addDefinitionBuilderChangeListener" externalInfoUrl="https://pmd.github.io/pmd-6.47.0/pmd_rules_java_codestyle.html#uselessqualifiedthis" priority="3">
Useless qualified this usage in the same class.
</violation>
</file>
```

**Running PMD through:** *Maven*

**Comment by garydgregory** [comment]
You are correct, sorry about the noise.
